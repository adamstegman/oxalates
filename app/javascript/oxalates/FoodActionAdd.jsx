import React from 'react';
import PropTypes from 'prop-types';

import "./FoodActionAdd.scss";

export const FoodActionAdd = ({ onStartAddingFood }) => {
  return (
    <button type="button"
            className="food-action"
            onClick={onStartAddingFood}>
      Add
    </button>
  );
};

FoodActionAdd.propTypes = {
  onStartAddingFood: PropTypes.func.isRequired,
};
