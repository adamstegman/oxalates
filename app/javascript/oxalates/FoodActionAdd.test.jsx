import React from 'react';
import renderer from 'react-test-renderer';

import { FoodActionAdd } from './FoodActionAdd';

test('FoodActionAdd renders the add food button', () => {
  const component = renderer.create(
    <FoodActionAdd onStartAddingFood={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
