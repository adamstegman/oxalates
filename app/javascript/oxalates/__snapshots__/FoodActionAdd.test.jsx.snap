// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodActionAdd renders the add food button 1`] = `
<button
  className="food-action"
  onClick={[Function]}
  type="button"
>
  Add
</button>
`;
